### ISSUE TITLE
False positive "Use of moved value" with Tokio 

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#9180]
## Environment

* **IntelliJ Rust plugin version:** 0.4.177.4801-222-nightly
* **Rust toolchain version:** 1.64.0-nightly (1b57946a4 2022-08-03) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.2 (CL-222.3345.126)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

Originally posted in https://youtrack.jetbrains.com/issue/CPP-30119

## Problem description
I found a set of errors. Looks like they are related to each other, but feel free to transfer any of them in a separate issue. 

1. Here, all errors are the "use of moved value"

<img width="627" alt="image" src="https://user-images.githubusercontent.com/49211026/183918464-17123d9c-10c9-4982-b5ab-94a3fe4b05cf.png">

2. After removing `loop`, new warnings appear:

<img width="627" alt="image" src="https://user-images.githubusercontent.com/49211026/183918801-e882bc69-be72-4311-8601-db092db48125.png">

(with red underline still is "use of moved value")
 
3. Quick fixes for grey warnings fix everything. 
![image](https://user-images.githubusercontent.com/49211026/183922086-a750607a-69f9-4383-8e2a-ef16044b09dd.png)

4. Add `loop`, false positives appear again:
<img width="932" alt="image" src="https://user-images.githubusercontent.com/49211026/183922168-29aaea7c-9487-4ab5-a9bf-8053c724b7f9.png">

## Steps to reproduce
```rust
#[tokio::main]
async fn main() {
    let (mut handle1, mut handle2) = (tokio::spawn(async {}), tokio::spawn(async {}));
    loop {
        handle1.abort();
        handle1.await;
        handle2.abort();
        handle2.await;
        (handle1, handle2) = (tokio::spawn(async {}), tokio::spawn(async {}));
    }
}
```

Cargo.toml:
```toml
[dependencies]
tokio = { version = "1.20.1", features = ["full"] }
```
